docs/Conformance-Guide-draft.md

```diff
@@ -0,0 +1,103 @@
+
+1. (Best Practice) Error message consistency: **For a consistent user experience error and log messages should follow the format documented in the Zowe Explorer extensibility documentation.**
+
+   To verify this requirement it is sufficient to provide links to user documentation showing screen shots of error dialogs as well as sample log file entries.
+
+1. (Best Practice) Zowe SDK usage: **Utilize available Zowe SDKs to standardize on z/OS interactions as well as other common capabilities used by many other Zowe extensions and tools.**
+
+   This can be verified by reviewing the dependencies list of the implementation such as in the package.json file of a nodejs-based extension.
+
+## Profiles Access extension conformance criteria (1)
+
+A Profiles Access extension is a Zowe Explorer extension that uses the Zowe Extensibility API to conveniently access Zowe CLI profiles loaded by Zowe Explorer itself. This allows the extension to consistently access profile instances of specific types, offer them for edit and updates as well as common refresh operations that apply to all extensions, add more profile types it is using itself for its own custom views (for example a CICS extension adding a CICS explorer view) and other similar use cases related to Zowe CLI profiles. These extensions do **not** have to be VS Code extension if it just wants to use ProfilesCache implementation of Zowe Explorer as all APIs are provided free of any VS Code dependencies. Such an extension could be used for another non-VS Code tool, a Zowe CLI plugin, a Web Server or another technology. However, to access the profiles cache of the actual running VS Code Zowe Explorer the extender needs to be a VS Code extension that has an extension dependency defined to be able to query the extender APIs. Therefore, some of the criteria that are listed here as required are only required if the extender is a VS Code extension.
+
+1. (Required if VS Code) VS Code extension dependency: **Zowe Explorer VS Code extensions can only be activated after Zowe Explorer is fully activated itself. Therefore, to ensure the correct activation order and extender must include an extensionDependencies to Zowe Explorer entry in their package.json file.**
+
+   The package.json file must be provided to verify this criteria.
+
+   ```json
+   "extensionDependencies": [
+    "Zowe.vscode-extension-for-zowe"
+   ],
+   ```
+
+1. (Required if VS Code) Zowe Extender access: **Zowe Explorer VS Code extensions must access the shared Zowe Explorer profiles cache only via the `ZoweExplorerApi.IApiRegisterClient.getExplorerExtenderApi()` API as documented in the Zowe Explorer extensibility documentation.**
```

Comment:
Extension accesses the shared Zowe Explorer profiles cache via ZoweExplorerApi.IApiRegisterClient.getExplorerExtenderApi() API as documented in the Zowe Explorer extensibility documentation.